core/src/test/java/org/apache/accumulo/core/rpc/ThriftUtilTest.java

```diff
@@ -0,0 +1,119 @@
+  }
+
+  @Test
+  public void testAccumuloTFramedTransportFactory() throws TTransportException {
+
+    // This test confirms that our custom FramedTransportFactory sets the max
+    // message size and max frame size to the value that we want.
+
+    TByteBuffer underlyingTransport = new TByteBuffer(ByteBuffer.allocate(1024));
+
+    AccumuloTFramedTransportFactory factory = new AccumuloTFramedTransportFactory(GB);
+    TTransport framedTransport = factory.getTransport(underlyingTransport);
+
+    assertEquals(framedTransport.getConfiguration().getMaxFrameSize(), GB);
+    assertEquals(framedTransport.getConfiguration().getMaxMessageSize(), GB);
+  }
+
+  @Test
+  public void testMessageSizeReadWriteSuccess() throws Exception {
+
+    // This test creates an 10MB buffer in memory as the underlying transport, then
+    // creates a TFramedTransport with a 1MB maxFrameSize and maxMessageSize. It then
+    // writes 1MB - 4 bytes (to account for the frame header) to the transport and
+    // reads the data back out.
+
+    TByteBuffer underlyingTransport = new TByteBuffer(ByteBuffer.allocate(MB10));
+    AccumuloTFramedTransportFactory factory = new AccumuloTFramedTransportFactory(MB1);
+    TTransport framedTransport = factory.getTransport(underlyingTransport);
+    assertEquals(framedTransport.getConfiguration().getMaxFrameSize(), MB1);
+    assertEquals(framedTransport.getConfiguration().getMaxMessageSize(), MB1);
+
+    byte[] writeBuf = new byte[MB1 - FRAME_HDR_SIZE];
+    Arrays.fill(writeBuf, (byte) 1);
+    framedTransport.write(writeBuf);
+    framedTransport.flush();
+
+    assertEquals(MB1, underlyingTransport.getByteBuffer().position());
+    underlyingTransport.flip();
+    assertEquals(0, underlyingTransport.getByteBuffer().position());
+    assertEquals(MB1, underlyingTransport.getByteBuffer().limit());
+
+    byte[] readBuf = new byte[MB1];
+    framedTransport.read(readBuf, 0, MB1);
+  }
+
+  @Test
+  public void testMessageSizeWriteFailure() throws Exception {
+
+    // This test creates an 10MB buffer in memory as the underlying transport, then
+    // creates a TFramedTransport with a 1MB maxFrameSize and maxMessageSize. It then
+    // writes 1MB + 100 bytes to the transport, which fails as it's larger than the
+    // configured frame and message size.
+
+    TByteBuffer underlyingTransport = new TByteBuffer(ByteBuffer.allocate(MB10));
+    AccumuloTFramedTransportFactory factory = new AccumuloTFramedTransportFactory(MB1);
+    TTransport framedTransport = factory.getTransport(underlyingTransport);
+    assertEquals(framedTransport.getConfiguration().getMaxFrameSize(), MB1);
+    assertEquals(framedTransport.getConfiguration().getMaxMessageSize(), MB1);
+
+    // Unable to write more than 1MB to the TByteBuffer
+    byte[] writeBuf = new byte[MB1 + 100];
+    Arrays.fill(writeBuf, (byte) 1);
+    framedTransport.write(writeBuf);
+    assertThrows(TTransportException.class, () -> framedTransport.flush());
```

Comment:
This method isn't actually testing the frame size... the exception being thrown is one about the read buffer being too small. It doesn't seem related to the frame size. So, I'm not sure this test is testing what it needs to test.

Reply:
I realized that my MB1, MB10, and MB100 variables were the same value due to copy/paste. I have fixed this and modified the test to confirm that the read does fail. Will push commit shortly.